src/sdg/services/management/commands/import_data_from_services.py

```diff
@@ -33,8 +35,44 @@ def handle(self, **options):
                     datum_check=datetime.fromisoformat(product["laatstGecheckt"]),
                 )
 
+        self.clean_upn_data()
+
         self.stdout.write(
             self.style.SUCCESS(
                 f"Successfully updated {updated_objects} localized generic products."
             )
         )
+
+    def clean_upn_data(self):
+        """
+        Delete the related Generic Product objects that do not match the UPN requirements:
+
+        - If there are SDG-codes: delete the Generic Product that does not have doelgroep.
+        - If there are no SDG-codes: delete the Generic Product that have a doelgroep.
+
+        * If there are any related products they must be deleted manually.
+        """
+        upn_qs = UniformeProductnaam.objects.all()
+        upn_qs.prefetch_related(
+            "generieke_producten",
+            "generieke_producten__producten",
+        )
+        for upn in upn_qs:
+            if upn.sdg:
+                generic_products = [
+                    i for i in upn.generieke_producten.all() if not i.doelgroep
+                ]
+            else:
+                generic_products = [
+                    i for i in upn.generieke_producten.all() if i.doelgroep
+                ]
+
+            for generic_product in generic_products:
+                if generic_product.producten.all():
+                    self.stdout.write(
+                        self.style.WARNING(
+                            f"Generic product '{generic_product}' has related products. Delete manually."
+                        )
+                    )
+                else:
+                    generic_product.delete()
```

Comment:
In retrospect, we should move this entire cleanup logic to either `sdg.core.management.utils.py` (around line 155) or `sdg.producten.management.commands.autofill.py` (around line 36).

The above 2 files now both do something with creating `GeneriekProduct` and then now in this command they are cleaned again. It should all be in 1 location/command.